sys/kern/src/arch/arm_m.rs

```diff
@@ -936,6 +936,17 @@ static TICKS: [AtomicU32; 2] = {
 #[no_mangle]
 pub unsafe extern "C" fn SysTick() {
     crate::profiling::event_timer_isr_enter();
+
+    // We don't need the current task pointer in this routine, but we'll access
+    // it briefly to update the stack watermark:
+    {
```

Comment:
i  wonder if this whole block should be conditional on the feature flag? I see that `update_stack_watermark` is a nop if the feature isn't enabled, and the compiler is *probably* smart enough to eliminate the `CURRENT_TASK_PTR` access when the function that's called on it doesn't do anything, but... I just kinda don't trust it...

Reply:
I was trying to make as little code conditional as possible, so that it always gets compiled; this is why it's only the body of update_watermark that's cfg'd. The compiler does appear to optimize this load out, fwiw.